### PR TITLE
Feature/analytics course preview

### PR DESCRIPTION
**Pull request by rostikjoystick**
**YouTrack task**: [#APPS-2826](https://vyahhi.myjetbrains.com/youtrack/issue/APPS-2826)

**Description List**:
* Added `Course preview screen opened` event to Amplitude analytics

### CODE REVIEW

File: app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt [hunk 1]
```diff
@@ -125,7 +128,15 @@ constructor(
             .subscribeOn(backgroundScheduler)
             .subscribeBy(
                 onComplete = { state = CourseView.State.EmptyCourse },
-                onSuccess  = { state = CourseView.State.CourseLoaded(it); postCourseViewedNotification(it.courseId) },
+                onSuccess  = {
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt:131]: Will it send event on every swipe to refresh?

Comment (rostikjoystick) [reply app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt:131]: Moved logging to `ScreenManagerImpl`

File: app/src/main/java/org/stepic/droid/core/ScreenManagerImpl.java [hunk 2]
```diff
@@ -231,6 +236,12 @@ public void showCourseModules(Context context, @NotNull Course course) {
 
     @Override
     public void showCourseScreen(Context context, @NotNull Course course, boolean autoEnroll, CourseScreenTab tab) {
+        analytic.reportAmplitudeEvent(AmplitudeAnalytic.CoursePreview.COURSE_PREVIEW_SCREEN_OPENED, new HashMap<String, Object>() {{
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/core/ScreenManagerImpl.java:239]: It seems that we lost deep link analytics

File: app/src/main/java/org/stepic/droid/core/ScreenManagerImpl.java [hunk 3]
```diff
@@ -92,6 +92,7 @@
 import java.io.File;
 import java.net.URLEncoder;
 import java.util.ArrayList;
+import java.util.HashMap;
```
Comment (eadm) [inline app/src/main/java/org/stepic/droid/core/ScreenManagerImpl.java:95]: unused import

File: app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt [hunk 4]
```diff
@@ -31,6 +33,7 @@ import org.stepik.android.view.injection.solutions.SolutionsSentBus
 import ru.nobird.android.presentation.base.PresenterBase
 import ru.nobird.android.presentation.base.PresenterViewContainer
 import ru.nobird.android.presentation.base.delegate.PresenterDelegate
+import timber.log.Timber
```
Comment (eadm) [inline app/src/main/java/org/stepik/android/presentation/course/CoursePresenter.kt:36]: logs should be removed

**Pull request merged**